core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -331,11 +315,86 @@ export class PreferencesPageComponent {
       this.loaderService.hideLoadingScreen();
     });
 
-    this.subjectInterestsChangeAtLeastOnce = false;
     this.TAG_REGEX_STRING = '^[a-z ]+$';
     this.LANGUAGE_CHOICES = this.languageUtilService.getLanguageIdsAndTexts();
     this.SITE_LANGUAGE_CHOICES = AppConstants.SUPPORTED_SITE_LANGUAGES;
   }
+
+  savePreferences(): void {
+    if (!this.hasUnsavedChanges) {
+      return;
+    }
+    this.alertsService.clearWarnings();
+    let emailData: EmailPreferencesBackendDict = {
+      can_receive_email_updates: this.canReceiveEmailUpdates,
+      can_receive_editor_role_email: this.canReceiveEditorRoleEmail,
+      can_receive_feedback_message_email: this.canReceiveFeedbackMessageEmail,
+      can_receive_subscription_email: this.canReceiveSubscriptionEmail
+    };
+    let updates = [
+      {
+        update_type: 'profile_picture_data_url',
+        data: this.newProfilePictureDataUrl
+      },
+      { update_type: 'user_bio', data: this.userBio },
+      { update_type: 'default_dashboard', data: this.defaultDashboard },
+      { update_type: 'subject_interests', data: this.subjectInterests },
+      {
+        update_type: 'preferred_language_codes',
+        data: this.preferredLanguageCodes
+      },
+      {
+        update_type: 'preferred_site_language_code',
+        data: this.preferredSiteLanguageCode
+      },
+      {
+        update_type: 'preferred_audio_language_code',
+        data: this.preferredAudioLanguageCode
+      },
+      { update_type: 'email_preferences', data: emailData }
+    ];
+    updates = updates.filter((update) => {
+      return this.unsavedChanges[update.update_type] === true;
+    });
+
+    if (AssetsBackendApiService.EMULATOR_MODE) {
+      // Remove 'profile_picture_data_url' from updates if the emulator mode is
```

Comment:
![Screenshot from 2024-01-25 23-45-11](https://github.com/oppia/oppia/assets/119884665/7fea8436-79c3-4514-bdf4-ea4ffaadb049)
As seen in the earlier code in the above image, if user is in emulator mode the file is saved to local storage, else it is posted to backend server.

Reply:
Oh, I see. But could you check why it's not actually possible to save in the backend in emulator mode? Offhand I don't see a reason why we can't do that, and if we can get rid of this branching then the code will be easier to maintain.

Reply:
@AkashPaloju You haven't replied to this comment.

Reply:
I have tried uploading profile image after removing 'if'-branches but it not worked. I have checked the code thoroughly in exploration editor handlers, imageUploadHandler, file_services, storage_services etc.. and this profile picture logic in backend but could not find the reason till now and investigating further.

I guess the developer who crafted this code had a compelling rationale behind incorporating this 'if' branching.

Reply:
Can you find the PR which introduced this? We should not assume.

Reply:
I guess it would be better if we open a new issue for this as this PR has not introduced this problem. What do you say?

Reply:
> Can you find the PR which introduced this? We should not assume -- if you don't know the "compelling rationale", let's find it out.

This [PR#16521](https://github.com/oppia/oppia/pull/16521/files#diff-0178d27d866bd962c1b96dd07b0e3a0cc8216714a3c8344cea73c3e4853ced38)  introduced this.

Hello, @lkbhitesh07  Please let us know why the profile-image is uploaded to local storage instead of GCS here , unlike in exploration creator page?
Thanks!

Reply:
Hey @AkashPaloju, You can always assign the person you want to take the next action otherwise that person might not get notified.
Regarding your question, I do not exactly remember why we went ahead with storing the images in the local storage. Me and my mentor were having some discussion around it and we agreed upon saving the images in local storage for the DEV mode. Now come to think of it we could've used the `dev_mode_storage_services.py` file to store the data in the backend, I think it would've worked.
Thanks

Reply:
@AkashPaloju Thanks for investigating. Based on Hitesh's response, I think it's fine to file an issue for this, and put a TODO in the code pointing to that issue. That would suffice -- thanks!

Reply:
Done. Added TODO comment and filed issue #19737